### PR TITLE
Add support for scraping Understat

### LINKED ISSUE #151: Incorporating Understat as a Data Source
[Understat](https://understat.com/) exposes data, current and historical, without any apparent restrictions on usage (i.e lacking a robots.txt) on the big 5. It has detailed data on both the player and club level, and even an [unofficial API](https://github.com/amosbastian/understat) that could perhaps be integrated as a data source.

### PR DESCRIPTION
**Pull request by JanVanHaaren**
This pull request adds support to scrape advanced statistics such as xG, xGBuildup and xGChain, and shot events with their associated xG values from the Understat website.

Concretely, this pull request includes the following changes.
* Extends the `BaseRequestsReader` class with functionality to extract JavaScript variables from the Understat website.
* Adds the `Understat` class with functionality to scrape leagues, seasons, schedules, team-match statistics, player-match statistics, player-season statistics and shot events.
* Adds documentation for the `Understat` class.

This pull request closes #151.

### CODE REVIEW

File: soccerdata/understat.py [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+                        "shots": _as_int(player["shots"]),
+                        "key_passes": _as_int(player["key_passes"]),
+                        "yellow_cards": _as_int(player["yellow_cards"]),
+                        "red_cards": _as_int(player["red_cards"]),
+                        "xg_chain": _as_float(player["xGChain"]),
+                        "xg_buildup": _as_float(player["xGBuildup"]),
+                    }
+                )
+
+        index = ["league", "season", "player"]
```
Comment (probberechts) [inline soccerdata/understat.py:437]: Can you add "team" to the index for consistency with other methods and data sources?

**Review by probberechts — CHANGES_REQUESTED** [review]
Thanks @JanVanHaaren. This is a great PR! The only issue I could find is a missing index level.

File: soccerdata/understat.py (hunk 1 above)
Comment (probberechts) [reply soccerdata/understat.py]: Fixed it myself :)

Comment (JanVanHaaren) [reply soccerdata/understat.py]: Thanks!

**Review by probberechts — APPROVED** [review]

**Pull request merged**